File: problems/14-dynamic-programming/001-house-robber/java/Solution.java
public class Solution {
    public int rob(int[] nums) {
        int twoBack = 0;
        int oneBack = 0;

        for (int money : nums) {
            int current = Math.max(oneBack, twoBack + money);
            twoBack = oneBack;
            oneBack = current;
        }

        return oneBack;
    }
}

File: problems/14-dynamic-programming/001-house-robber/python/solution.py
class Solution:
    def rob(self, nums: list[int]) -> int:
        two_back = 0
        one_back = 0

        for money in nums:
            current = max(one_back, two_back + money)
            two_back = one_back
            one_back = current

        return one_back
